### PR TITLE
build(gha): Add workflow for backend linting

### PR DESCRIPTION
**Pull request by billyvg**
Adds a workflow for backend linting

### CODE REVIEW

**Review by joshuarli — CHANGES_REQUESTED** [review]
Could you also modify some python files in a way that black and flake8 would not agree with, so we can see it happening? After the suggested changes.

File: .github/workflows/backend-lint.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+jobs:
+  lint:
+    name: backend lint
+    runs-on: ubuntu-16.04
+    timeout-minutes: 10
+    env:
+      MIGRATIONS_TEST_MIGRATE: 1
+
+    steps:
+      - uses: actions/checkout@v1
```
[the remark this answers is not in the document]
Comment (billyvg) [reply .github/workflows/backend-lint.yml:16]: Hmm I wonder if this will fail our `git diff` checking for modified files

**Comment by billyvg** [comment]
@joshuarli Changed to use `paths-filter` action and you can see the failing run here: https://github.com/getsentry/sentry/runs/1345222169

**Comment by joshuarli** [comment]
> @joshuarli Changed to use `paths-filter` action and you can see the failing run here: https://github.com/getsentry/sentry/runs/1345222169
> 
> Also made a repl to test these changes

In the future we might have like, yamllint in pre-commit, but for now I guess we're treating it to be python-specific. So is it possible to also add path filters in addition to `created|modified`? It's expanding to `pre-commit run --files '.github/workflows/backend-lint.yml' 'src/sentry/conf/server.py'`.

**Review by joshuarli — APPROVED** [review]
lgtm, just: https://github.com/getsentry/sentry/pull/21740#issuecomment-720883717

**Pull request merged**